src/include/migraphx/raw_data.hpp

```diff
@@ -284,10 +313,19 @@ bool operator==(const T& x, const U& y)
     bool result   = x.empty() and y.empty();
     if(not result and xshape == yshape)
     {
-        visit_all(x, y)([&](auto xview, auto yview) { result = xview == yview; },
-                        [&](auto&& xs, auto&& ys) {
-                            result = std::equal(xs.begin(), xs.end(), ys.begin(), ys.end());
-                        });
+        if(xshape.computable())
+        {
+            visit_all(x, y)([&](auto xview, auto yview) { result = xview == yview; },
+                            [&](auto&& xs, auto&& ys) {
+                                result = std::equal(xs.begin(), xs.end(), ys.begin(), ys.end());
+                            });
+        }
+        else
+        {
+            // not tuple type so do not need tuple visitor
+            x.raw_visit(
```

Comment:
Nesting visits is usually not good, but then we would need a `raw_visit_all` function.

Reply:
Well in this case, its not a visit, so its not a problem, but we should probably not call it `raw_visit` in the first place.